src/ctapipe/monitoring/calculator.py

```diff
@@ -0,0 +1,337 @@
+        self.outlier_detectors = {}
+        if self.outlier_detector_list is not None:
+            for outlier_detector in self.outlier_detector_list:
+                self.outlier_detectors[
+                    outlier_detector["apply_to"]
+                ] = OutlierDetector.from_name(
+                    name=outlier_detector["name"],
+                    validity_range=outlier_detector["validity_range"],
+                    subarray=self.subarray,
+                    parent=self,
+                )
+
+
+class StatisticsCalculator(CalibrationCalculator):
+    """
+    Component to calculate statistics from calibration events.
+
+    This class inherits from ``CalibrationCalculator`` and is responsible for
+    calculating various statistics from calibration events, such as pedestal
+    and flat-field data. It aggregates statistics, detects outliers,
+    handles faulty data chunks.
+    The ``StatisticsCalculator`` holds two functions to conduct two different passes
+    over the data with and without overlapping chunks. The first pass is conducted
+    with non-overlapping, while overlapping chunks can be set by the ``chunk_shift``
+    parameter for the second pass. The second pass over the data is only conducted
+    in regions of trouble with a high percentage of faulty pixels exceeding
+    the threshold ``faulty_pixels_threshold``.
+    """
+
+    chunk_shift = Int(
+        default_value=None,
+        allow_none=True,
+        help=(
+            "Number of samples to shift the aggregation chunk for the calculation "
+            "of the statistical values. Only used in the second_pass(), since the "
+            "first_pass() is conducted without overlapping chunks (chunk_shift=None)."
+        ),
+    ).tag(config=True)
+
+    faulty_pixels_threshold = Float(
+        default_value=10.0,
+        allow_none=True,
+        help=(
+            "Threshold in percentage of faulty pixels over the camera "
+            "to identify regions of trouble."
+        ),
+    ).tag(config=True)
+
+    def first_pass(
+        self,
+        table,
+        tel_id,
+        masked_pixels_of_sample=None,
+        col_name="image",
+    ) -> Table:
+        """
+        Calculate the monitoring data for a given set of events with non-overlapping aggregation chunks.
+
+        This method performs the first pass over the provided data table to calculate
+        various statistics for calibration purposes. The statistics are aggregated with
+        non-overlapping chunks (``chunk_shift`` set to None), and faulty pixels are detected
+        using a list of outlier detectors.
+
+
+        Parameters
+        ----------
+        table : astropy.table.Table
+            DL1-like table with images of shape (n_images, n_channels, n_pix), event IDs and
+            timestamps of shape (n_images, )
+        tel_id : int
+            Telescope ID for which the calibration is being performed
+        masked_pixels_of_sample : ndarray, optional
```

Comment:
Perhaps, you can consider using `MaskedColumn` columns in the table?

Reply:
At the moment, I do not have the the info about "bad pixels" in the LST-1 files. Therefore, I used as an optional ndarray and passed `None`. If we change the type here to `MaskedColumn` and one point down in the aggregation we would need to access the data via `masked_pixels_of_sample.data` which would break for `NoneType`.